services/lumberjack/src/service.js

```diff
@@ -0,0 +1,255 @@
+    ]);
+
+    logger.debug('Service stopped.');
+  }
+
+  // Create the koa api and return the http server.
+  async _createApi(influx) {
+    const app = new Koa();
+
+    app.use(koaLogger());
+
+    app.context.influx = influx;
+
+    // Set up the router middleware.
+    app.use(router.routes());
+    app.use(router.allowedMethods());
+
+    // Start and wait until the server is up and then return it.
+    return await new Promise((resolve, reject) => {
+      const server = app.listen(this._port, (err) => {
+        if (err) reject(err);
+        else resolve(server);
+      });
+
+      // Add a promisified close method to the server.
+      server.closeAsync = () => new Promise((resolve) => {
+        server.close(() => resolve());
+      });
+    });
+  }
+
+  _startTasks() {
+    this._pingTask =
+      createTimeoutTask(this._pingTask.bind(this), this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._uploadRateTask =
+      createTimeoutTask(this._uploadRate.bind(this), this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._telemetryTask =
+      createTimeoutTask(this._telemetryOverview.bind(this),
+        this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+  }
+
+  /** Get service ping data and write to the database */
+  async _pingTask() {
+    // Get ping data
+    let ping =
+      (await request.get('http://' + this._pingHost + ':' +
+        this._pingPort + '/api/ping')
+        .proto(stats.PingTimes)
+        .timeout(1000)).body;
+
+    // Write data for services
+    for (let endpoint of ping.service_pings) {
+      let { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: 'lumberjack'
+      });
+    }
+
+    // Write data for devices
+    for (let endpoint of ping.device_pings) {
+      let { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { devicePing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: 'lumberjack'
+      });
+    }
+  }
+
+  /** Get telemetry upload rate data and write to the database */
+  async _uploadRate() {
+    // Get upload rate
+    let rate =
+      (await request.get('http://' + this._forwardInteropHost + ':' +
+        this._forwardInteropPort + '/api/upload-rate')
+        .proto(stats.InteropUploadRate)
+        .timeout(1000)).body;
+
+    await this._influx.writeMeasurement('upload-rate', [
+      {
+        fields: { t1: rate.total_1, t5: rate.total_5,
+          f1: rate.fresh_1, f5: rate.fresh_5 },
+        tags: { host: this._forwardInteropHost, port: this._forwardInteropPort }
+      }]);
+  }
+
+  /** Get telemetry overview and task queue length and
+  write to the database */
+  async _telemetryOverview() {
```

Comment:
There should be two telemetry tasks started, one that communications with the plane telemetry, and one with ground telemetry.